core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala

```diff
@@ -112,6 +113,15 @@ abstract class FileCommitProtocol {
    * just crashes (or killed) before it can call abort.
    */
   def abortTask(taskContext: TaskAttemptContext): Unit
+
+  /**
+   * Specifies that a file should be deleted with the commit of this job. The default
+   * implementation deletes the file immediately, but this may be overriden to delay the physical
+   * deletion of the file until commit time.
```

Comment:
is this api extension going open source soon? if yes, we probably don't want to document it this way (just remove everything after ", but")